app/access/api.py

```diff
@@ -55,14 +58,20 @@ def create_user(request: HttpRequest, user_in: UserSchema) -> UserSchema:
         - 404 (Not Found) if there is no provider with the given provider ID
         - 409 (Conflict) if there is already a record with the same username
         - 422 (Unprocessable Content) if there is any other invalid value
+        - 500 (Internal Server Error) if there is inconsistency with cognito
+        - 503 (Service Unavailable) if cognito cannot be reached
     """
     provider = get_object_or_404(Provider, id=user_in.provider_id)
 
-    user_out = User.objects.create(
-        username=user_in.username,
-        first_name=user_in.first_name,
-        last_name=user_in.last_name,
-        email=user_in.email,
-        provider=provider
-    )
+    with transaction.atomic():
+        user_out = User.objects.create(
+            username=user_in.username,
+            first_name=user_in.first_name,
+            last_name=user_in.last_name,
+            email=user_in.email,
+            provider=provider
+        )
+        created = create_user(user_out)
+        if not created:
+            raise HttpError(500, "Internal Server Error")
```

Comment:
I think we should log this, ideally with a reference to the call (like adding the username in the log message) so we can trace it down.

Reply:
Nevermind, just realized that the utils.creat_user does that